source/plugin.js

```diff
@@ -0,0 +1,39 @@
+const { autoTasks } = require('./internals/tasks');
+const { rlInterface } = require('./internals/options');
+const { executeTask } = require('./internals/executor');
+
+module.exports = [
+  {
+    description: 'fully automated project state clean: like a freshly-cloned, never-started repo',
+    name: 'clean-project-auto',
+    func: () => {
+
+      console.log('');
+      console.log('Executing fully-automated project clean.');
+      console.log('Use \'react-native clean-project\' for more control');
+      console.log('Use \`./node_modules/.bin/react-native-clean-project\' for total control');
+      console.log('');
+
+      rlInterface.close(); // if we don't do this it hangs waiting for input
```

Comment:
The tests' results also raise a warning regarding an open task that's keeping preventing jest from exiting, and it's related to `rlInterface`. You can check it by running `./node_modules/.bin/jest --detectOpenHandles`. Can you give it a look, please?

Reply:
I gave that one a long look and I'm not sure how to fix it well. My general take was that it was like that before - by which I mean structurally you have the readline opened immediately whether it will be used or not, then you close it in an unrelated area. My conversion to a plugin style didn't relate to that though it did mean I had to be careful to close it myself in plugin.js. Unfortunately even though I'm closing it there it isn't closing when run by jest.

It was late enough at night and I'd worked enough on it that after contemplating the "no fix" path (no real consequence) I let it go. It's an existing issue with the code that needs cleaning but it'll require re-organizing how readline is used in the options.js which are orthogonal to my change.

Can we let it go for this PR?

Reply:
You also have a point here and I'll take it in another PR. Thank you for explaining it thoroughly.